DFE-Digital/sts-plan-technology-for-your-school
# fix: add app environment variable, amend E2E cookies page, fix E2E artifact naming

## Overview

- Fixes issue with an E2E test in staging, where the staging environment won't have a particular element but all other environments will.
- Amends E2E artifact output names to fix conflicts (231355)

## Changes

### Minor

- Adds environment variable to E2E tests for what environment the app is running on
- Add a check for the above variable to the failing test on staging
- Append the environment, and current datetime, to artifact names to avoid artifact name conflicts

## How to review the PR

1. Run E2E tests against local web app with 0 changes; ensure E2E tests work
2. Set the environment variable `APP_ENVIRONMENT` to "staging"
3. Verify that this test gets skipped in the `cookies-page.cy.js` tests:

```javascript
      cy.get('meta[name="google-site-verification"]').should('exist');
```

I also tested setting the environment variable to _not_staging, and amending the check to `should('not.exist')` to ensure it failed, and then changing environment variable back to staging and ensuring it didn't fail.

## Additional notes

We should consolidate these:

[e2e-tests.yml workflow](/.github/workflows/e2e-tests.yml)
[run-e2e-tests action.yml](/.github/actions/run-e2e-tests/action.yml)

## Checklist

Delete any rows that do not apply to the PR.
